src/v2/DevTools/setupTestWrapper.tsx

```diff
@@ -68,14 +72,22 @@ type SetupTestWrapper<T extends OperationType> = {
  * @see https://relay.dev/docs/en/testing-relay-components
  * @see https://testing-library.com/docs/react-testing-library/
  */
+
+type RTLRenderResult = RenderResult<
+  typeof import("@testing-library/dom/types/queries"),
+  HTMLElement
+>
+type RenderWithRelay = RTLRenderResult & { env: RelayMockEnvironment }
 export const setupTestWrapperTL = <T extends OperationType>({
   Component,
   query,
   variables = {},
 }: SetupTestWrapper<T>) => {
-  const renderWithRelay = (mockResolvers: MockResolvers = {}) => {
+  const renderWithRelay = (
+    mockResolvers: MockResolvers = {},
+    manualEnvControl?: boolean
```

Comment:
I added the option to control the env from the caller, when we need to check for refetches we can't in the current structure and exporting the env alone runs into racing conditions to mock the operations.

Reply:
@araujobarret - if there's an example of this somewhere in the codebase min posting back here in a comment? That way there's a bit of an example / papertrail that other devs can refer to

Reply:
@damassi this was the first idea introduced here, before with enzyme we were doing the strange [spy thing](https://github.com/artsy/force/blob/main/src/v2/Apps/Fair/Components/__tests__/FairBooths.jest.tsx#L62-L69)

The approach I did here I based on the [docs](https://relay.dev/docs/guides/testing-relay-components/#refetch-component) how to test refetch case, but in my case here I only cared about the `operation.request`, I didn't remock the fixture based on the filter change, maybe we can refactor another test(without a big fixture 🙏 ) with RTL + refetch test idea and put in the docs as reference, I don't know a good one to do it, maybe this `FairBooths`